content/news/2023-07-07-bevy-0.11/index.md

```diff
@@ -620,6 +620,34 @@ impl ViewNode for BloomNode {
 }
 ```
 
+## `#[reflect(default)]` on Enum Variant Fields
+
+<div class="release-feature-authors">authors: @MrGVSV</div>
+
+When using the `FromReflect` trait, fields marked `#[reflect(default)]` will be set to their `Default` value if they don't exist on the reflected object.
+
+Previously, this was only supported on struct fields. 
+Now, it is also supported on all enum variant fields.
+
+```rust
+#[derive(Reflect)]
+enum MyEnum {
+    Data {
+        #[reflect(default)]
+        a: u32,
+        b: u32,
+    },
+}
+
+let mut data = DynamicStruct::default ();
+data.insert("b", 1);
+
+let dynamic_enum = DynamicEnum::new("Data", data);
+
+let my_enum = MyEnum::from_reflect( & dynamic_enum).unwrap();
+assert_eq!(u32::default(), my_enum.a);
+```
```

Comment:
This is perhaps a bit overkill. We could just cut this down to the enum definition with the `#[reflect(default)]` and remove the other stuff. Or we could even cut the code entirely (though it's nice to see what the change is in terms of code).